test/e2e/framework/pod/resource.go

```diff
@@ -386,49 +386,54 @@ func isNotRestartAlwaysMirrorPod(p *v1.Pod) bool {
 	return p.Spec.RestartPolicy != v1.RestartPolicyAlways
 }
 
-// NewExecPodSpec returns the pod spec of hostexec pod
-func NewExecPodSpec(ns, name string, hostNetwork bool) *v1.Pod {
+// NewAgnhostPod returns a pod that uses the agnhost image. The image's binary supports various subcommands
+// that behave the same, no matter the underlying OS.
+func NewAgnhostPod(ns, podName string, volumes []v1.Volume, mounts []v1.VolumeMount, ports []v1.ContainerPort, args ...string) *v1.Pod {
 	immediate := int64(0)
 	pod := &v1.Pod{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      name,
+			Name:      podName,
 			Namespace: ns,
 		},
 		Spec: v1.PodSpec{
 			Containers: []v1.Container{
-				{
-					Name:            "agnhost",
-					Image:           imageutils.GetE2EImage(imageutils.Agnhost),
-					ImagePullPolicy: v1.PullIfNotPresent,
-				},
+				NewAgnhostContainer("agnhost-container", mounts, ports, args...),
 			},
-			HostNetwork:                   hostNetwork,
+			Volumes:                       volumes,
 			SecurityContext:               &v1.PodSecurityContext{},
 			TerminationGracePeriodSeconds: &immediate,
 		},
 	}
 	return pod
 }
 
+// NewAgnhostContainer returns the container Spec of an agnhost container
```

Comment:
Might be useful to mention in the comment that if no args are supplied then it defaults to running the pause command

Reply:
I think it might be a good idea to consider renaming the args parameter to agnHostCommand so that as to make it clear that a agnhost command is being chosen here.

Add a see also : https://github.com/kubernetes/kubernetes/tree/master/test/images/agnhost#agnhost to the comment

Reply:
Adding the comment about the pause command is a good idea, will do that.

As for ``agnhostCommand``, I don't think it fits, and it could potentially be confused with the Pod's Command.

Furthermore, there are a few tests which do not rely on the agnhost binary at all, but instead they run something else. So, because of this, I think just ``args`` is ok.